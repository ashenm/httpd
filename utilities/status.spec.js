const status = require('./status');

describe('Function success', () => {
  it('Should return well-formed response when message explicitly specified', () => {
    expect(status.success(200, 'Hooray!')).toMatchSnapshot();
  });

  it('Should return well-formed response with a default status text when message unspecified', () => {
    expect(status.success(200)).toMatchSnapshot();
  });
});

describe('Function failure', () => {
  it('Should return well-formed response when message explicitly specified', () => {
    expect(status.failure(400, 'Boo!')).toMatchSnapshot();
  });

  it('Should return well-formed response with a default status text when message unspecified', () => {
    expect(status.failure(400)).toMatchSnapshot();
  });
});
