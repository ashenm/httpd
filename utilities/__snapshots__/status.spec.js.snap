// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function failure Should return well-formed response when message explicitly specified 1`] = `
Object {
  "error": Object {
    "code": 400,
    "message": "Boo!",
  },
}
`;

exports[`Function failure Should return well-formed response with a default status text when message unspecified 1`] = `
Object {
  "error": Object {
    "code": 400,
    "message": "Bad Request",
  },
}
`;

exports[`Function success Should return well-formed response when message explicitly specified 1`] = `
Object {
  "status": Object {
    "code": 200,
    "message": "Hooray!",
  },
}
`;

exports[`Function success Should return well-formed response with a default status text when message unspecified 1`] = `
Object {
  "status": Object {
    "code": 200,
    "message": "OK",
  },
}
`;
